addons/mail/static/src/components/file_uploader/file_uploader_tests.js

```diff
@@ -87,6 +89,48 @@ QUnit.test('no conflicts between file uploaders', async function (assert) {
     );
 });
 
+QUnit.test('does not crash when an attachment is removed before its upload starts', async function (assert) {
+    assert.expect(1);
+
+    const attachmentUploadedPromise = makeTestPromise();
+    await this.start({
+        async mockFetch(resource, init) {
+            const _super = this._super.bind(this, ...arguments);
+            if (resource === '/web/binary/upload_attachment') {
+                await attachmentUploadedPromise;
```

Comment:
Missing comment for `attachmentUploadProm` purpose (btw would rename to be shorter). Short enough to not be noisy, but also good to have it to ease test readability at a glance.